lib/engines/hcktest/tests.rb

```diff
@@ -45,6 +45,13 @@ def list_tests(log: false)
       retry
     end
 
+    def load_tests_results
+      @tests_results = @tools.list_all_results(@target['key'], @client.name, @tag)
+
+      @last_done_results = done_test_results
+      @new_done_tests = @last_done_results - @last_handled_done_results
```

Comment:
To save instance variables:

```suggestion
      last_done_results = done_test_results
      @tests_results = @tools.list_all_results(@target['key'], @client.name, @tag)
      @new_done_tests = last_done_results - done_test_results
```

Reply:
`@new_done_tests = last_done_results - done_test_results` - NO

new_done_tests is  a diff between last_done_results and last_handled_done_results. We set last_handled_done_results in `def handle_finished_test_results` after handling new_done_tests. 

`@new_done_tests = last_done_results - done_test_results` is a diff between 2 calls of list_all_results, but this does check handle we it or not.

In case, we manually rerun several tests via VNC, we can have a raise condition new results exist but are not processed. That is the reason, why I set `last_handled_done_results` only after processing results.

Reply:
We can rename `load_tests_results` to reflect that it advances the iteration. `@test_results` at the beginning of this method will represent the previous iteration, and the updated value will represent the next iteration. We can model the state change of the loop this way.

Reply:
`update_tests_results`?

Reply:
That sounds good to me.